master/buildbot/www/graphql.py

```diff
@@ -65,12 +68,36 @@ def writeError(msg, errcode=400):
 
         return self.asyncRenderHelper(request, self.asyncRender, writeError)
 
-    def renderQuery(self, query):
+    async def renderQuery(self, query):
         query = self.graphql.parse(query)
         errors = self.graphql.validate(self.schema, query)
         if errors:
             raise Error(400, [e.formatted for e in errors])
 
+        async def field_resolver(parent, resolve_info, **args):
```

Comment:
I am still not very happy with this implementation.
I think it is still a bit too hacky. I have not yet found where to manage this, but probably this needs to be implemented in an independent module